src/components/OrderItemCard.vue

```diff
@@ -0,0 +1,129 @@
+        <p v-if="$filters.getFeature(getProduct(item.productId).featureHierarchy, '1/SIZE/')"> {{ $t("Size") }}: {{ $filters.getFeature(getProduct(item.productId).featureHierarchy, '1/SIZE/') }} </p>
+      </ion-label>
+      <div class="metadata">
+        <StatusBadge :statusDesc="item.orderItemStatusDesc || ''" :key="item.orderItemStatusDesc"/>
+        <StatusBadge v-if="item.facilityId === orderPreOrderFacilityId || item.facilityId === orderBackOrderFacilityId" :statusDesc="item.facilityId === orderPreOrderFacilityId ? 'Pre-order' : 'Backorder' || ''" :key="item.facilityName"/>
+      </div>
+    </ion-item>
+    <!-- TODO: Need to handle this property -->
+    <div v-if="item.facilityId === orderPreOrderFacilityId || item.facilityId === orderBackOrderFacilityId">
+      <ion-item>
+        <ion-label>{{ $t("Promise date") }}</ion-label>
+        <p slot="end"> {{ item.promisedDatetime ? $filters.formatUtcDate(item.promisedDatetime, 'YYYY-MM-DDTHH:mm:ssZ', 'D MMM YYYY') : '-'  }} </p>
+      </ion-item>
+      <ion-item>
+        <ion-label>{{ $t("PO arrival date") }}</ion-label>
+        <!-- TODO: Need to handle this property -->
+        <p slot="end"> {{ item.promiseOrderArrivalDate ? $filters.formatUtcDate(item.promiseOrderArrivalDate, 'YYYY-MM-DDTHH:mm:ssZ', 'D MMM YYYY') : '-' }} </p>
+      </ion-item>
+      <ion-item>
+        <ion-label>{{ $t("Parking") }}</ion-label>
+        <!-- TODO: Need to handle this property -->
+        <p slot="end"> {{ item.facilityName ? item.facilityName : '-' }} </p>
+      </ion-item>
+    </div>
+    <div v-else>
+      <ion-item>
+        <ion-label>{{ $t("Shipping method") }}</ion-label>
+        <p slot="end"> {{ item.shipmentMethodTypeId ? getShipmentMethodDesc(item.shipmentMethodTypeId) : '-' }} </p>
+      </ion-item>
+      <div v-if="item.facilityId !== '_NA_' && item.orderItemStatusId !== 'ITEM_COMPLETED'">
+        <ion-item>
+          <ion-label>{{ $t("Shipping from") }}</ion-label>
+          <!-- Used ion-badge to display a dot after the facility name when the ship from facility is brokering queue -->
+          <p slot="end">
+            {{ item.facilityName ? item.facilityName : "-" }}
+          </p>
+        </ion-item>
+        <ion-item lines="none">
+          <ion-label>{{ $t("Location inventory") }}</ion-label>
+          <p slot="end">{{ getProductStock(item.productId) }}</p>
+        </ion-item>
+      </div>
+      <div v-if="item.facilityId === '_NA_' && item.orderItemStatusId !== 'ITEM_COMPLETED'">
+        <ion-item>
+          <ion-label>{{ $t("Auto cancel") }}</ion-label>
+          <p slot="end">{{ item.autoCancelDate ? moment.utc(item.autoCancelDate).fromNow() : '-' }}</p>
+        </ion-item>
+        <ion-item lines="none">
+          <ion-label>{{ $t("Parking") }}</ion-label>
+          <p slot="end">{{ 'Brokering queue' }}</p>
+        </ion-item>
+      </div>
+      <div v-if="item.orderItemStatusId === 'ITEM_COMPLETED'">
```

Comment:
Implement a method for the condition. It will help when we need such checks at multiple places